Sage/carbon
# feat(select): add poperjs positioning mechanism - FE-2546

1. This PR adds Popperjs positioning to `Select` components (including `Filterable` and `MultiSelect`)
2. This PR also fixes an incosistent close behaviour of `MultiSelect` component when deleting selected options while the list is open (compare `controlled` and `default` `MultiSelect` stories on carbon.sage.com)
3. This PR also refactors `Popover` component and adds `ResizeObserver` to allow popperjs to react to dimension change of the reference element.
 
Fixes #2582 
Fixes #3695

### Proposed behaviour

![Zrzut ekranu 2021-02-12 o 09 43 36](https://user-images.githubusercontent.com/20651022/107746916-e98d6d00-6d16-11eb-9395-caa2c2aeab16.png)
![Zrzut ekranu 2021-02-12 o 09 40 56](https://user-images.githubusercontent.com/20651022/107746929-ee522100-6d16-11eb-94b3-8a3f56f367d2.png)

### Current behaviour
![Zrzut ekranu 2021-02-12 o 09 43 17](https://user-images.githubusercontent.com/20651022/107746865-d7abca00-6d16-11eb-86fd-bb9af40ede75.png)
![Zrzut ekranu 2021-02-12 o 09 40 28](https://user-images.githubusercontent.com/20651022/107746880-dd091480-6d16-11eb-9dff-b156893537fd.png)

### Checklist




### Testing instructions
Check `design-system-flat-table--paginated` story for the similar scenario to the one described in #2582 
Check `Select`, `MultiSelect` and `FilterableStories` if the automatic positioning works
Check `MultiSelect` component and confirm that closing behaviour is consistent across all the stories
Due to the refactor of `Popover` component it is worth to also check `ActionPopover`, `SplitButton` and `MultiActionButton` because all these components use `Popover` internally
